src/otx/core/metrics/fmeasure.py

```diff
@@ -780,8 +786,141 @@ def classes(self) -> list[str]:
         return self.label_info.label_names
 
 
+class MeanAveragePrecisionFMeasure(MeanAveragePrecision):
```

Comment:
Is there any reason to implement this from scratch? I think there is another way to exploit https://lightning.ai/docs/torchmetrics/stable/pages/overview.html#torchmetrics.MetricCollection. For example,

```
class MeanAveragePrecisionFMeasure(MetricCollection):
     def __init__(self, **kwargs):
          super().__init__([MeanAveragePrecision(), FMeasure(**kwargs)])
     def compute(self, best_confidence_threshold: float | None = None) -> dict:
```